### PR TITLE
Add Sweep Name in the UI and resolve some bugs 

### PR DESCRIPTION
**Pull request by tchaton**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning-hpo/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?

Fixes # (issue).

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: lightning_training_studio/controllers/sweep.py [hunk 1]
```diff
@@ -47,11 +48,12 @@ def on_reconcile_start(self, sweeps: List[SweepConfig]):
...
                     self.tensorboard_sweep_id.append(id)
                     self.db.insert(TensorboardConfig(sweep_id=id, shared_folder=str(drive.drive_root)))
-                elif sweep.stage in (Stage.FAILED, Stage.SUCCEEDED):
+                elif sweep.stage in (Stage.FAILED, Stage.SUCCEEDED) and id not in self.tensorboard_sweep_id_detach:
                     for tensorboard in self.db.select_all(TensorboardConfig):
                         if tensorboard.sweep_id == id:
-                            tensorboard.desired_stage = Stage.DELETED
+                            tensorboard.desired_stage = Stage.STOPPED
                             self.db.update(tensorboard)
+                            self.tensorboard_sweep_id_detach.append(id)
```
Comment (tchaton) [inline lightning_training_studio/controllers/sweep.py:56]: Fix a bug where the Sweep Controller would keep stopping the Tensorboard instance.

File: lightning_training_studio/controllers/tensorboard.py [hunk 2]
```diff
@@ -26,7 +27,9 @@ def on_reconcile_start(self, configs: List[TensorboardConfig]):
                         cloud_compute=L.CloudCompute("cpu-small"),
                     )
                     self.r[work_name].stage = Stage.PENDING
-            if config.desired_stage == Stage.DELETED:
+            elif config.desired_stage == Stage.STOPPED:
+                self._stop_tensorboard(work_name)
```
Comment (tchaton) [inline lightning_training_studio/controllers/tensorboard.py:31]: Cleaned up stopping on tensorboard. The spec wasn't properly updated.

File: lightning_training_studio/framework/pytorch_lightning.py [hunk 3]
```diff
@@ -159,8 +160,14 @@ def on_train_batch_end(self, trainer, pl_module, *args) -> None:
                     human_readable = get_human_readable_count(total_parameters)
                     self.work.total_parameters = str(human_readable)
 
-                if trainer.checkpoint_callback.last_model_path:
-                    self.work.last_model_path = Path(trainer.checkpoint_callback.last_model_path)
+                ckpt = trainer.checkpoint_callback
+
+                if ckpt.best_model_score and ckpt.best_model_score != self.work.best_model_score:
```
Comment (tchaton) [inline lightning_training_studio/framework/pytorch_lightning.py:165]: Collect best model score live.

File: lightning_training_studio/controllers/tensorboard.py [hunk 4]
```diff
@@ -18,6 +18,7 @@ class TensorboardController(Controller):
     def on_reconcile_start(self, configs: List[TensorboardConfig]):
         for config in configs:
             work_name = urllib.parse.quote_plus(config.sweep_id)
+            print(work_name, work_name not in self.r, config)
```
Comment (lantiga) [inline lightning_training_studio/controllers/tensorboard.py]:
```suggestion
```

File: tests/controllers/test_sweep.py [hunk 5]
```diff
@@ -57,12 +57,14 @@ def test_sweep_controller(monkeypatch):
     response = sweep_controller.run_sweep(config)
     assert response == "The current Sweep 'a' is running. It couldn't be updated."
 
+    counter = 0
```
Comment (lantiga) [inline tests/controllers/test_sweep.py:60]:
```suggestion
```

File: tests/controllers/test_sweep.py [hunk 6]
```diff
@@ -57,12 +57,14 @@ def test_sweep_controller(monkeypatch):
...
     while True:
         sweep: Sweep = sweep_controller.r.get("a", None)
         sweep.data = None
         sweep_controller.run("a", "b")
         if sweep.stage == Stage.SUCCEEDED:
             break
+        counter += 1
```
Comment (lantiga) [inline tests/controllers/test_sweep.py:67]:
```suggestion
```

**Review by lantiga — APPROVED** [review]
Great, just a few nits

File: tests/controllers/test_sweep.py (hunk 5 above)
Comment (tchaton) [reply tests/controllers/test_sweep.py:60]: Removed.

**Pull request merged**